infrastructure/messaging/application/middleware/logging_middleware.py

```
@@ -0,0 +1,59 @@
+                    "path": request.url.path,
+                    "status_code": response.status_code,
+                    "process_time_ms": round(process_time * 1000, 2)
+                }
+            )
+
+            response.headers["X-Request-ID"] = request_id
+            return response
+
+        except Exception as e:
+            process_time = time.time() - start_time
+            logger.error(
```

Ruff (0.15.11)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)